File: skeleton/src/main/java/me/shkschneider/skeleton/helpers/SystemHelper.java
package me.shkschneider.skeleton.helpers;

import android.os.SystemClock;

public class SystemHelper {

    public static String uname() {
        return String.format("%s %s %s", SystemProperties.osName(), SystemProperties.osVersion(), SystemProperties.osArch());
    }

    public static long sinceBoot() {
        return SystemClock.elapsedRealtime();
    }

    public static long sinceCurrentThreadBirth() {
        return SystemClock.currentThreadTimeMillis();
    }

    public static void safeSleep(final long milliseconds) {
        SystemClock.sleep(milliseconds);
    }

}

File: android/src/main/kotlin/me/shkschneider/skeleton/android/core/helper/SystemHelper.kt
package me.shkschneider.skeleton.android.core.helper

import android.os.SystemClock
import me.shkschneider.skeleton.android.core.helperx.SystemProperties

// <http://developer.android.com/reference/android/content/Context.html>
object SystemHelper {

    fun uname(): String {
        return String.format(LocaleHelper.Device.locale(),
                "%s %s %s",
                System.getProperty(SystemProperties.OS_NAME),
                System.getProperty(SystemProperties.OS_VERSION),
                System.getProperty(SystemProperties.OS_ARCH))
    }

    fun sinceBoot(): Long {
        // Returns milliseconds since boot, including time spent in sleep.
        return SystemClock.elapsedRealtime()
    }

    fun uptime(): Long {
        // Returns milliseconds since boot, not counting time spent in deep sleep.
        return SystemClock.uptimeMillis()
    }

    @Suppress("DeprecatedCallableAddReplaceWith")
    @Deprecated("Does not return until at least the specified number of milliseconds has elapsed.") // Avoid
    fun sleep(milliseconds: Long) {
        SystemClock.sleep(milliseconds)
    }

}
